### DISCUSSION TITLE
Support for credentialless iframes to address COEP header needs

### DISCUSSION THREAD

**Discussion by coatless** [quarto-dev/quarto-cli#4913] [Feature Requests]
Right now, the video short code `{{ video }}` provides a standard iframe that automatically assumes the pages default headers.

https://github.com/quarto-dev/quarto-cli/blob/b0d668e4287f74f5208b510f37414a3341d7973b/src/resources/extensions/quarto/video/video.lua#L81

https://github.com/quarto-dev/quarto-cli/blob/b0d668e4287f74f5208b510f37414a3341d7973b/src/resources/extensions/quarto/video/video.lua#L108

https://github.com/quarto-dev/quarto-cli/blob/b0d668e4287f74f5208b510f37414a3341d7973b/src/resources/extensions/quarto/video/video.lua#L124

Would it be possible to have the short-code support or set by default a `credentialless` state? e.g.

```
<iframe credentialless src="https://example.com">
```

The `credentialles` state is important for [Cross-Origin-Embedder-Policy
 (COEP)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Embedder-Policy) environments. For [webR](https://docs.r-wasm.org/webr/latest/serving.html) startup and package installs, we need to set the COOP and COEP headers to significantly speed up the availability of in the browser R editor. We're running into issues with the iframe because when we turn on COEP, then any embedded lecture video from YouTube using the video shortcode is blocked with `"youtube refused to connect."`

![](https://user-images.githubusercontent.com/833642/226725454-76ab386b-d5df-46ac-9f7b-b7ce89a188af.png)

By having the `<iframe>` tag include `credentialless`, the iframe is loaded from a different, empty context. In particular, it is loaded without cookies. This allows for the removal of the COEP restriction for just the video will allow the entire page to still fall under the custom COOP and COEP headers.

For more background, please see: 

- Background on `iframe-credentialless` set: <https://developer.chrome.com/blog/iframe-credentialless/>
- Initial Proposal: <https://github.com/camillelamy/explainers/blob/main/anonymous_iframes.md>

Where the issue arose: https://github.com/coatless/quarto-webr/issues/8

**Comment by cscheid** [comment]
This seems like something we should 100% support. I'll open an issue to track, thanks!

We're in hard-freeze mode for the upcoming 1.3 release, but we'll do this early in the 1.4 cycle, since it seems pretty low-risk.

**Reply by coatless** [reply]
Just a heads up, I think this is only supported on chromium-based browsers so far. 

I've gently poked the Mozilla folks for their position: 

https://github.com/mozilla/standards-positions/issues/628

**Comment by ghost** [comment]
Thanks for reporting this @coatless and the great supporting research.  Let's move the conversation to #4914.
